R/begin.R

```diff
@@ -16,14 +17,29 @@
 begin = function(
   wd = NULL, 
   load = c( 'magrittr', 'dplyr' ),
+  keep = NULL,
   scipen = FALSE,
   verbose = TRUE,
   repos = 'http://cran.us.r-project.org'
 ){
   
   # clear workspace.
-  rm( list = ls( all.names = TRUE, envir = parent.frame() ), envir = parent.frame() )
-  if( verbose ) cat( 'Cleared variables from environment. \n' )
+  envobjs = ls( all.names = TRUE, envir = parent.frame() )
+  drop = setdiff( envobjs, keep )
+  # check if attempting to keep any objects not in environment
+  if( length( keep ) > 0 ){
+    missing = setdiff( keep, envobjs )
+    if( length( missing ) > 0 ){ 
+      stop( 
+        glue::glue('
+          Attempted to keep objects not in the environment: [{cc(missing, sep = ", ")}]. Error easyr::begin E911.
+      ')) 
+    }
+  }
+
+  # clear objects not set to keep
+  rm( list = drop, envir = parent.frame() )
+  if( verbose ){ cat( 'Cleared objects not in \"keep\" argument from environment. \n' ) }
```

Comment:
FWIW \" is not necessary here. If you started with one quote type (') then you can use the other (") without escaping (and vice-versa). No need to fix it though.